### ISSUE TITLE
deno.json workspace support for Deno 2

### ISSUE THREAD

**Issue by dsherret** [denoland/deno#22942]
Tracking issue for unstable workspaces:

- [x] https://github.com/denoland/deno/issues/22020
- [x] https://github.com/denoland/deno/issues/21769
- [x] https://github.com/denoland/deno/issues/18546
- [x] Make properties in deno.json work well based on top level workspace
- [x] Make it work well with npm packages
- [ ] Update documentation https://github.com/denoland/deno-docs/issues/415
- [x] #24163 
- [x] #24162 
- [x] #24161 
- [x] #24160 
- [x] #24159 
- [x] https://github.com/denoland/deno_config/issues/55

**Comment by marvinhagemeister** [comment]
Adding for completeness: I think the setup in Fresh is a bit odd given that the website is located _inside_ the folder structure of Fresh itself.

```sh
<repo>/  # <- Fresh source here
  www/ # <- inside fresh/ folder
```

I would assume that we would switch to a each having their own directory when moving to workspaces:

```sh
<repo>/
  /fresh
  /www
```

The more important thing for me is if shared dependencies are properly hoisted to the top level workspaces folder to make deduplication work properly.

**Comment by bartlomieju** [comment]
When implementing this we should move as much code as possible to `deno_config` so other tools like `dnt` can reuse it.

**Comment by dsherret** [comment]
Closing this as this will be released in Deno 1.45. Still pending documentation, but that will be completed later today.

**Comment by igl** [comment]
Amazing! Can't wait to try it out
